File: app/src/main/java/com/fractalwrench/crazycats/resource/StringResourceFileReader.java
package com.fractalwrench.crazycats.resource;


import com.fractalwrench.crazycats.common.PreConditions;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;

/**
 * Reads a resource file as a string
 */
class StringResourceFileReader {

    String readResourceAsString(String resName) throws IOException {
        PreConditions.checkNonNull(resName);
        InputStream inputStream = getClass().getClassLoader()
                                            .getResourceAsStream(resName);

        if (inputStream == null) {
            throw new IOException(String.format("Failed to find resource '%s'", resName));
        }

        try {
            BufferedReader br = new BufferedReader(new InputStreamReader(inputStream));

            String line;
            StringBuilder sb = new StringBuilder();

            while ((line = br.readLine()) != null) {
                sb.append(line);
            }
            return sb.toString();
        }
        finally {
            inputStream.close();
        }
    }

}

File: app/src/main/java/com/fractalwrench/crazycats/resource/StringResourceFileReader.kt
package com.fractalwrench.crazycats.resource


import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader

/**
 * Reads a resource file as a string
 */
internal class StringResourceFileReader {

    @Throws(IOException::class)
    fun readResourceAsString(resName: String): String {
        val inputStream = javaClass.classLoader
                .getResourceAsStream(resName) ?: throw IOException(String.format("Failed to find resource '%s'", resName))

        inputStream.use { inputStream ->
            val br = BufferedReader(InputStreamReader(inputStream))

            var line: String?
            val sb = StringBuilder()

            line = br.readLine()

            while (line != null) {
                sb.append(line)
                line = br.readLine()
            }
            return sb.toString()
        }
    }

}
